quicklisp/client.lisp

```diff
@@ -77,13 +80,50 @@
 (defun help ()
   "For help with Quicklisp, see http://www.quicklisp.org/beta/")
 
-(defun uninstall (system-name)
+(defun uninstall (systems &key remove-dependencies)
+  (unless (consp systems)
+    (setf systems (list systems)))
+  (dolist (system-name systems systems)
+    (let ((system (find-system system-name)))
+      (cond ((and system remove-dependencies)
+             (uninstall-system-dependencies system-name))
+            (system
+             (ql-dist:uninstall system))
+            (t
+             (warn "Unknown system ~S" system-name)
+             nil)))))
+
+(defun uninstall-system-dependencies (system-name)
```

Comment:
I'd prefer to see this factored out a bit to produce the list of candidates to uninstall without actually uninstalling them. That will make it easier to test.